### PR TITLE
Docs/refactor button story

### PR DESCRIPTION
**Pull request by ErikssonJoakim**
This PR implements #459 for `Button`

Recap of the most important changes:

- conformed descriptions to format (bold **values**, `properties` in back tics, default values in story description).
- Cleaned up redundant code (unused deprecated imports, redundant meta parameters).
- added default values in Overview story args to improve property table, as per [wiki](https://github.com/okp4/ui/wiki/Create-A-Story) specifications.
- ordered stories as they appear in property table.
- Replaced the story Content by stories `label`, `icon` and `left / right icon`.
Capture d’écran 2022-09-05 à 14 27 55
Capture d’écran 2022-09-05 à 14 27 20
- Added interactivity to `onClick` story.
Capture d’écran 2022-09-05 à 14 26 39

### CODE REVIEW

**Review by JoeyDjr — APPROVED** [review]

**Review by JoeyDjr — APPROVED** [review]
It's better with the Toasts !

File: stories/atoms/components/button/Button.stories.mdx [hunk 1]
```diff
@@ -62,155 +67,204 @@ that will help users to find the action they're looking for.
...
-
-### Properties
+## Properties
 
 <ArgsTable story="Overview" />
 
+## Label
+
+The `label` is the text description of **Button** aswell as its [title](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/title).
+If **Button** is an **icon** `variant` the `label` is not shown, however it is still there as a title attribut used for the [tooltip](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Tooltip_Role).
```
Comment (lolottetheclash) [inline stories/atoms/components/button/Button.stories.mdx:77]: Maybe you could space out the text a bit by skipping a line?
Moreover, I think that the addition of the links on title and tooltip add an additional formatting and I have the impression that the reading is complicated between the text in bold, the code one, plus the links in blue. But this is just my opinion!

File: stories/atoms/components/button/Button.stories.mdx [hunk 2]
```diff
@@ -62,155 +67,204 @@ that will help users to find the action they're looking for.
...
   </Story>
 </Canvas>
 
 ## Background colors
 
-The background color of the `Button` component is declined according to the selected theme and allows to better mark a user action.
+**Button** have a palette of background colors to choose from and allows to better mark a user action.
+The background color can be a visual aid as to what type of action the button executes.
+
+By _default_ `backgroundColor` is **primary**.
```
Comment (lolottetheclash) [inline stories/atoms/components/button/Button.stories.mdx:161]: After reflection, I think we should not put "default" in italics, there are too many different texts for me. What do you think about it ?

File: stories/atoms/components/button/Button.stories.mdx [hunk 3]
```diff
@@ -62,155 +67,204 @@ that will help users to find the action they're looking for.
...
+              justifyContent: 'center'
+            }}
+          >
+            <Button label="Click me!" onClick={handleState('button clicked', true)} />
+          </div>
+          <Toast
+            description={
+              <div style={{ display: 'flex', flexDirection: 'column' }}>
+                <Typography color="highlighted-text" fontSize="small" fontWeight="light">
+                  {`Like this component? See `}
```
Comment (lolottetheclash) [inline stories/atoms/components/button/Button.stories.mdx:247]: Very nice idea !!!

**Review by lolottetheclash — COMMENTED** [review]
The refactor is great, but I let you some minor remarks to discuss with the others :)

File: stories/atoms/components/button/Button.stories.mdx [hunk 4]
```diff
@@ -29,27 +27,26 @@ import { Icon } from 'ui/atoms/icon/Icon'
...
----
+## Overview
 
 <Story
   name="Overview"
   argTypes={{
-    icon: { control: false, table: { type: { summary: 'JSX.Element' } } },
-    leftIcon: { control: false, table: { type: { summary: 'JSX.Element' } } },
-    rightIcon: { control: false, table: { type: { summary: 'JSX.Element' } } }
+    icon: { control: false, table: { type: { summary: 'Node' } } },
```
Comment (fredericvilcot) [inline stories/atoms/components/button/Button.stories.mdx:47]: Regarding the `icon` property, I can't see a dedicated story to explain how it works..

File: stories/atoms/components/button/Button.stories.mdx (hunk 1 above)
Comment (fredericvilcot) [reply stories/atoms/components/button/Button.stories.mdx:77]: Fully agree with @lolottetheclash , we need to offer a more readable documentation by limiting the use of multiple text formats.

File: stories/atoms/components/button/Button.stories.mdx [hunk 5]
```diff
@@ -70,155 +67,204 @@ that will help users to find the action they're looking for.
...
+      <Button backgroundColor="primary" variant="primary" label="Medium" size="medium" />
+      <Button backgroundColor="primary" variant="primary" label="Large" size="large" />
+    </div>
   </Story>
 </Canvas>
 
 ## Background colors
 
-The background color of the `Button` component is declined according to the selected theme and allows to better mark a user action.
+**Button** have a palette of background colors to choose from and allows to better mark a user action.
```
Comment (fredericvilcot) [inline stories/atoms/components/button/Button.stories.mdx:158]:
```suggestion
**Button** has a palette of background colors to choose from and allows to better mark a user action.
```

File: stories/atoms/components/button/Button.stories.mdx [hunk 6]
```diff
@@ -70,155 +67,204 @@ that will help users to find the action they're looking for.
...
-`leftIcon` and `rightIcon`.
+## OnClick
 
-- `label` is the only obligatory property and is used to clearly indicate what action will be triggered, through the
-  tooltip and text.
-- `icon` sets the icon for the **variant** 'icon'.
-- `leftIcon` and `rightIcon` are auxilliary elements giving great freedom to embellish the button
-  together with the label.
+`onClick` is what makes the magic happen, without it the button has no corresponding action.
+The function passed as value should correspond to the action described in `label`.
```
Comment (fredericvilcot) [inline stories/atoms/components/button/Button.stories.mdx:224]: I do not understand this sentence..
What is the link between the callback function and the action described in `label`?

File: stories/atoms/components/button/Button.stories.mdx [hunk 7]
```diff
@@ -70,155 +67,204 @@ that will help users to find the action they're looking for.
...
-        <Button
-          label="Add wallet"
-          leftIcon={<Icon name="wallet" size={22} />}
-          rightIcon={<Icon name="add" size={22} />}
-        />
-      </div>
-    )}
+  <Story name="OnClick">
+    {() => {
+      const [state, setState] = useState({ name: '', isOpened: false })
```
Comment (fredericvilcot) [inline stories/atoms/components/button/Button.stories.mdx:229]: Why the use of an object as state? 
As you only handle one `Toast`, I will lighten the code: 
```suggestion
      const [opened, setOpened] = useState(false)
```
I let you implement the rest of the modifications 😌

**Review by fredericvilcot — CHANGES_REQUESTED** [review]
That's great, but we can go much deeper on the refactor! 
Don't hesitate to ask for help or for more precisions 😌

File: stories/atoms/components/button/Button.stories.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ErikssonJoakim) [reply stories/atoms/components/button/Button.stories.mdx]: That's a good idea!

File: stories/atoms/components/button/Button.stories.mdx (hunk 6 above)
Comment (ErikssonJoakim) [reply stories/atoms/components/button/Button.stories.mdx:225]: I have changed the description and removed the mention of `label`.

**Review by lolottetheclash — COMMENTED** [review]
Nice refactor, I wrote you some remarks that we can discuss afterwards :)

File: stories/atoms/components/button/Button.stories.mdx [hunk 8]
```diff
@@ -31,196 +28,341 @@ import { Icon } from 'ui/atoms/icon/Icon'
...
+          />
+        </div>
+      )
+    }}
+  </Story>
+</Canvas>
+
+## Icon
+
+`Icon` allows for the creation of an icon button by passing the icon of your choice. It simply takes any node element as a value.
```
[the remark this answers is not in the document]
Comment (ErikssonJoakim) [reply stories/atoms/components/button/Button.stories.mdx:119]: I prefer to keep `for` as a filler word. It's true it's not needed for the comprehension but I find that the sentence is easier to read with it.

File: stories/atoms/components/button/Button.stories.mdx [hunk 9]
```diff
@@ -31,196 +28,341 @@ import { Icon } from 'ui/atoms/icon/Icon'
...
-
-### Properties
+## Properties
 
 <ArgsTable story="Overview" />
 
+## Label
+
+The `label` is the text description of Button as well as its [title](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/title).
+If Button is an **icon** `variant` the `label` is not shown, however it is still there as a title attribut used for the [tooltip](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Tooltip_Role).
```
[the remark this answers is not in the document]
Comment (ErikssonJoakim) [reply stories/atoms/components/button/Button.stories.mdx:85]: Thanks 🙂 These are good improvements. I will see how I can improve the first sentence.

**Review by lolottetheclash — APPROVED** [review]
Well done !

File: stories/atoms/components/button/Button.stories.mdx [hunk 10]
```diff
@@ -31,196 +28,343 @@ import { Icon } from 'ui/atoms/icon/Icon'
...
----
-
-### Properties
+## Properties
 
 <ArgsTable story="Overview" />
 
+## Label
+
+The `label` is the text description of Button that help communicate what happens after the click.
```
Comment (fredericvilcot) [inline stories/atoms/components/button/Button.stories.mdx:84]:
```suggestion
The `label` is the text description of Button that helps communicate what happens after the click.
```

File: stories/atoms/components/button/Button.stories.mdx [hunk 11]
```diff
@@ -31,196 +28,343 @@ import { Icon } from 'ui/atoms/icon/Icon'
...
+attribut used for the [tooltip](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Tooltip_Role)
+(as you can see by hovering your mouse over the icon button below).
+
+The `label` can have any length however Button width will never exceed 273px.
+
+<Canvas>
+  <Story name="Label">
+    {() => {
+      const { theme } = useTheme()
+      const invert = theme === 'light'
```
Comment (fredericvilcot) [inline stories/atoms/components/button/Button.stories.mdx:95]: If we want to be consistent about the naming of boolean variables, I would have named it:
```suggestion
      const isLightTheme = theme === 'light'
```

File: stories/atoms/components/button/Button.stories.mdx [hunk 12]
```diff
@@ -31,196 +28,343 @@ import { Icon } from 'ui/atoms/icon/Icon'
...
+          />
+        </div>
+      )
+    }}
+  </Story>
+</Canvas>
+
+## Icon
+
+`Icon` allows for the creation of an icon button by passing the icon of your choice. It simply takes any node element as a value.
```
Comment (fredericvilcot) [inline stories/atoms/components/button/Button.stories.mdx:121]:
```suggestion
`Icon` turns the button into a more concise, unlabeled element. For that, you can pass the icon of your choice as a value.
```

File: stories/atoms/components/button/Button.stories.mdx [hunk 13]
```diff
@@ -31,196 +28,343 @@ import { Icon } from 'ui/atoms/icon/Icon'
...
+
+Requires Button to be an **icon** `variant`, without it the button content will stay `label`.
```
Comment (fredericvilcot) [inline stories/atoms/components/button/Button.stories.mdx:123]:
```suggestion
Requires Button to be an **icon** `variant`. Without it, the button content will remain `label`.
```

File: stories/atoms/components/button/Button.stories.mdx [hunk 14]
```diff
@@ -31,196 +28,343 @@ import { Icon } from 'ui/atoms/icon/Icon'
...
+
+<Canvas>
+  <Story name="Icon">
+    {() => {
+      const { theme } = useTheme()
+      const invert = theme === 'light'
```
Comment (fredericvilcot) [inline stories/atoms/components/button/Button.stories.mdx:129]: Same remark as above about boolean variables naming.

File: stories/atoms/components/button/Button.stories.mdx [hunk 15]
```diff
@@ -31,196 +28,343 @@ import { Icon } from 'ui/atoms/icon/Icon'
...
 
-The `Button` component comes with three different sizes to allow its use according to its location within the interface.
+There are three different sizes to allow Button to be used according to its location within the interface.
 
-- **Small**: particularly suitable for small screens and must not saturate the space. This size should also be used for integrations in content spaces such as cards, footers, etc...
-- **Medium**: makes it possible to adapt to a greater number of uses.
-- **Large**: particularly suitable for large screens or to embed icons befor or after.
+- **Small**: particularly suitable for small screens and must not saturate the space. This size should also be used for integrations in content spaces such as cards, footers, etc.
+- **Medium** (default): makes it possible to adapt to a greater number of uses.
+- **Large**: particularly suitable for large screens or in conjonction with `leftIcon` / `rightIcon`.
```
Comment (fredericvilcot) [inline stories/atoms/components/button/Button.stories.mdx:199]:
```suggestion
- **Large**: particularly suitable for large screens or in conjonction with `leftIcon` / `rightIcon` properties.
```

File: stories/atoms/components/button/Button.stories.mdx [hunk 16]
```diff
@@ -31,196 +28,343 @@ import { Icon } from 'ui/atoms/icon/Icon'
...
-        <Button
-          label="Add wallet"
-          leftIcon={<Icon name="wallet" size={22} />}
-          rightIcon={<Icon name="add" size={22} />}
-        />
-      </div>
-    )}
+  <Story name="OnClick">
+    {() => {
+      const [opened, setOpened] = useState(false)
```
Comment (fredericvilcot) [inline stories/atoms/components/button/Button.stories.mdx:305]: As I said above about boolean variables naming, we could be more consistent:
```suggestion
      const [isOpened, setIsOpened] = useState(false)
```

File: stories/atoms/components/button/Button.stories.mdx [hunk 17]
```diff
@@ -31,196 +28,343 @@ import { Icon } from 'ui/atoms/icon/Icon'
...
+      const handleOpened = isOpened => () => {
```
Comment (fredericvilcot) [inline stories/atoms/components/button/Button.stories.mdx:306]: I'm not a big fan of this syntax when parameter is not mandatory for the method.
I think we can be more concise:
```suggestion
      const handleOpen = () => {
        setIsOpened(prevState => !prevState)
```

**Review by fredericvilcot — CHANGES_REQUESTED** [review]
Thanks for this work @ErikssonJoakim, it's much more sexier! 😍
I wrote down some remarks that, IMHO, bring more readibility to the story 😌

**Comment by ErikssonJoakim** [comment]
@fredericvilcot Thanks for all remarks 😃 They really help improve the PR. FYI I've pushed the changes and I took the liberty to do minor changes to `variant` and `size` descriptions as well, so be aware of that 🥳

**Review by fredericvilcot — APPROVED** [review]
All good, thx @ErikssonJoakim!

**Review by JoeyDjr — APPROVED** [review]

**Pull request merged**